build.sh

```diff
@@ -220,6 +220,11 @@ if [[ -n "${APT_PROXY}" ]] && ! curl --silent "${APT_PROXY}" >/dev/null ; then
 	exit 1
 fi
 
+if [[ ${#WPA_PASSWORD} -lt 8 || ${#WPA_PASSWORD} -gt 63  ]] ; then
```

Comment:
Is this length check valid? Can the passphrase have escape characters, for example?

Reply:
From wikipedia the WPA pass-phrase should be"8 to 63 printable ASCII characters". https://w1.fi/cgit/hostap/tree/wpa_supplicant/wpa_passphrase.c checks that there are no control characters.

With this patch if the user does something tricksy like

`WPA_PASSWORD="abcd"$'\n'"efgh"`

Then it will pass this test (9 chars) and fail later on in wpa_passphrase, but at least the error message will be visible.

Options would be:

1) No check here, just let wpa_passphrase catch the  errors (other half of patch makes that visable).

2) Partial check/current patch. Just check length

3) Full check. Try to catch all errors early.